src/main/java/address/util/collections/ReorderedList.java

```diff
@@ -54,23 +59,34 @@ public synchronized int size() {
     }
 
     /**
-     * Moves the element in the list
-     * @param from The index(before shifting occurred) of the element to be shifted(before shift).
-     * @param to The index(before shifting occurred) of the list where element is to be shifted to.
+     * Moves the elements in the list.
+     * Precondition: The object at destinationIndex is not in the list of toMove.
+     * @param toMove The list of objects to be moved.
+     * @param destinationIndex The index(before shifting) of the list where elements are to be shifted to.
      */
-    public synchronized void moveElement(int from, int to) {
-
-        if (from < to) {
-            //Element to be shifted is below the index where the element need to be shifted to.
-            //Removing the element first will shift every element down.
-            //Therefore, only remove the element after shifting finished.
-            T tmpPerson = mappingList.get(from);
-            mappingList.add(to, tmpPerson);
-            mappingList.remove(from);
-        } else if (from > to) {
-            T tmpPerson = mappingList.remove(from);
-            mappingList.add(to, tmpPerson);
+    public synchronized Collection<Integer> moveElements(List<T> toMove, int destinationIndex) {
+
+        List<Integer> movedIndexes = new ArrayList<>();
+
+        if (destinationIndex == mappingList.size()){
```

Comment:
Comments here would be good, not easy to tell what kind of optimisation is happening here